src/main/scala/util/RunUtils.scala

```diff
@@ -573,7 +573,7 @@ object RunUtils {
     var iteration = 1
     var modified: Boolean = true
     val analysisResult = mutable.ArrayBuffer[StaticAnalysisContext]()
-    while (modified || (analysisResult.size < 2 && config.memoryRegions)) {
+    while (modified || (analysisResult.size < 2)) {
```

Comment:
We don't want to run the analyses multiple times unless it's necessary for the memory regions